modules/modes/ev_train.py

```diff
@@ -0,0 +1,207 @@
+from typing import Generator
+
+from rich.table import Table
+
+from modules.context import context
+from modules.map import get_map_data_for_current_position, get_map_data
+from modules.map_data import MapFRLG, MapRSE, PokemonCenter, get_map_enum
+from modules.map_path import calculate_path, PathFindingError
+from modules.modes import BattleAction
+from modules.player import get_player_avatar
+from modules.pokemon import get_party, get_opponent, StatusCondition, StatsValues
+from ._interface import BotMode, BotModeError
+from .util import navigate_to, heal_in_pokemon_center, spin
+from ..battle_state import BattleOutcome
+from ..battle_strategies import BattleStrategy, DefaultBattleStrategy
+from ..console import console
+from ..encounter import handle_encounter, EncounterInfo
+from ..gui.ev_selection_window import ask_for_ev_targets
+
+closest_pokemon_centers: dict[MapFRLG | MapRSE, list[PokemonCenter]] = {
+    # Hoenn
+    MapRSE.ROUTE101: [PokemonCenter.OldaleTown],
+    MapRSE.ROUTE102: [PokemonCenter.OldaleTown, PokemonCenter.PetalburgCity],
+    MapRSE.ROUTE103: [PokemonCenter.OldaleTown],
+    MapRSE.ROUTE104: [PokemonCenter.PetalburgCity, PokemonCenter.RustboroCity],
+    MapRSE.ROUTE116: [PokemonCenter.RustboroCity],
+    MapRSE.ROUTE110: [PokemonCenter.SlateportCity, PokemonCenter.MauvilleCity],
+    MapRSE.ROUTE117: [PokemonCenter.MauvilleCity, PokemonCenter.VerdanturfTown],
+    MapRSE.ROUTE113: [PokemonCenter.FallarborTown],
+    MapRSE.ROUTE114: [PokemonCenter.FallarborTown],
+    MapRSE.ROUTE119: [PokemonCenter.FortreeCity],
+    MapRSE.ROUTE120: [PokemonCenter.FortreeCity],
+    MapRSE.ROUTE121: [PokemonCenter.LilycoveCity],
+    # Kanto
+    MapFRLG.ROUTE1: [PokemonCenter.ViridianCity],
+    MapFRLG.ROUTE22: [PokemonCenter.ViridianCity],
+    MapFRLG.ROUTE2: [PokemonCenter.ViridianCity, PokemonCenter.PewterCity],
+    MapFRLG.ROUTE3: [PokemonCenter.PewterCity, PokemonCenter.Route4],
+    MapFRLG.ROUTE4: [PokemonCenter.Route4, PokemonCenter.CeruleanCity],
+    MapFRLG.ROUTE24: [PokemonCenter.CeruleanCity],
+    MapFRLG.ROUTE6: [PokemonCenter.VermilionCity],
+    MapFRLG.ROUTE11: [PokemonCenter.VermilionCity],
+    MapFRLG.ROUTE9: [PokemonCenter.Route10],
+    MapFRLG.ROUTE10: [PokemonCenter.Route10],
+    MapFRLG.ROUTE7: [PokemonCenter.CeladonCity],
+    MapFRLG.ROUTE18: [PokemonCenter.FuchsiaCity],
+}
+
+
+_list_of_stats = ("hp", "attack", "defence", "speed", "special_attack", "special_defence")
```

Comment:
Change this to be in the standard EV order
```suggestion
_list_of_stats = ("hp", "attack", "defence", "special_attack", "special_defence", "speed")
```